src/mongoc/mongoc-client.c

```diff
@@ -1183,6 +1184,7 @@ _mongoc_client_command_simple_with_hint (mongoc_client_t           *client,
                                    command, NULL, read_prefs);
 
    cursor->hint = hint;
+   cursor->is_write_command = true;
```

Comment:
I reasoned that we didn't need a private setter function, since the driver already has access to the cursor struct.